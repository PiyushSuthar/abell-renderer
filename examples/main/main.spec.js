const path = require('path');
const {
  buildAndGetSelector
} = require('../../tests/utils/baseTestFramework.js');

const TEST_MAP = [
  {
    desc: 'should test javascript inside brackets',
    query: '[data-test="js-add-check"]'
  },
  {
    desc: 'should test if reassignment of variable is possible',
    query: '[data-test="reassignment-check"]'
  },
  {
    desc: 'should check if values can be required from json and js',
    query: '[data-test="require-check"]'
  },
  {
    desc: 'should test if putting slash before bracket prints content as it is',
    query: '[data-test="comment-check"]'
  },
  {
    desc: 'should test if projects are correctly rendered',
    query: '[data-test="projects-check"]'
  }
];

describe('examples/main', () => {
  let $;
  beforeAll(async () => {
    $ = await buildAndGetSelector({
      exampleToRun: 'main',
      outPath: path.join(__dirname, 'out.html')
    });
  });

  for (const test of TEST_MAP) {
    it(test.desc, () => {
      expect($(test.query).html()).toMatchSnapshot();
    });
  }
});
