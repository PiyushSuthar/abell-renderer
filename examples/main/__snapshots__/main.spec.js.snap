// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/main should check if values can be required from json and js 1`] = `"69"`;

exports[`examples/main should test if index 0, has project name Project Two 1`] = `"Project Two"`;

exports[`examples/main should test if index 1, has project name Project One 1`] = `"Project One"`;

exports[`examples/main should test if index 2, has project name Project Three 1`] = `"Project Three"`;

exports[`examples/main should test if putting slash before bracket prints content as it is 1`] = `"{{ print this as it is }}"`;

exports[`examples/main should test if reassignment of variable is possible 1`] = `"9"`;

exports[`examples/main should test javascript inside brackets 1`] = `"11"`;
